### ISSUE TITLE
Sensor devices with floating point values or values greater 100

### ISSUE THREAD

**Issue by Elisedd-Bradan** [thkl/hap-homematic#412]
Hi, 

I'm wondering if it might be possible to integrate an air pressure sensor into the HomeKit. In fact I have a variable that contains the actual air pressure value in the CCU - how ever - I did't make it to show the value in the Homekit. The reason for this is, that I did't found a template that can transfer values out of the range from 1 - 100. Is this a limitation of the HomeKit Server itself or would it be possible to add a sensor template this can handle higher values - and even better also floating point values?

THX Andreas

**Comment by thkl** [comment]
As for my knowledge there are no air pressure measuring devices available in HomeKit. This is the list of supported devices: 
![image](https://user-images.githubusercontent.com/884355/113740392-f28c3080-9700-11eb-9175-996cdc2282f1.png)

**Comment by thkl** [comment]
Each HomeKit device can have different services  (this is the list above).
Each service has multiple characteristics (this is the list of all apple defined types) :

![image](https://user-images.githubusercontent.com/884355/113758970-d2ff0300-9714-11eb-86ea-ba37b2ea52d2.png)

As an example a thermostat device will use the thermostat service. Per Apples definition there will be a characteristic for the current temperature and the target temperature. 

You are able to define any device with any characteristic (Custom) or also your own characteristic (for example the air pressure):

![image](https://user-images.githubusercontent.com/884355/113759276-2e30f580-9715-11eb-8ed8-037f0aafe982.png)

But, the apple home app does not know anything about your custom device so it will not show data or controls for your device:

![image](https://user-images.githubusercontent.com/884355/113759505-705a3700-9715-11eb-9949-34997255933b.png)

Some 3rd party HomeKit apps, like eve,  act a little bit different and  will show "custom" devices. They just take the characteristics and will print them on the screen, or give the user a slider to change the values if the characteristics is writable.

![image](https://user-images.githubusercontent.com/884355/113759591-8536ca80-9715-11eb-9bd4-e88732246972.png)


Long story short: You are able to create a custom device to show the air pressure. But the Apple Home App will just present you a tile with "not supported" ...

**Comment by thkl** [comment]
Breaking ++++

Eve knows about Air Pressure. They use it in there eve weather devices. If I will choose the correct uuid it will be shown in eve like this:

![image](https://user-images.githubusercontent.com/884355/113760418-7ef51e00-9716-11eb-860e-6e343bb90c0b.png)

So i am able to implement this, but it will only work with the eve app.

**Comment by Elisedd-Bradan** [comment]
Whow - Perfect :) THX also for the explanation. Not - that I think that this will help fast - but I will also ask Apple to implement this characteristic, because this is really missing from my point of view and should be implemented in the air quality sensor service...

**Comment by thkl** [comment]
This will work in eve only with the next version 
 
![image](https://user-images.githubusercontent.com/884355/114263562-8e11ef80-99e6-11eb-9a50-9063d710732b.png)

![image](https://user-images.githubusercontent.com/884355/114264390-f4990c80-99ea-11eb-8976-bfb0b56ca7db.png)
